pkgs/development/python-modules/openmm/default.nix

```diff
@@ -0,0 +1,87 @@
+{ buildPythonPackage, lib, fetchFromGitHub, cmake, clang, gfortran, fftwSinglePrec
+, doxygen, ocl-icd, opencl-headers, swig, python, cython, numpy
+, cudatoolkit, enableCuda ? false
+} :
+
+buildPythonPackage rec {
+  pname = "openmm";
+  version = "7.7.0";
+
+  src = fetchFromGitHub {
+    owner = "openmm";
+    repo = pname;
+    rev = version;
+    hash = "sha256-2PYUGTMVQ5qVDeeABrwR45U3JIgo2xMXKlD6da7y3Dw=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    gfortran
+    swig
+    cython
+    doxygen
+  ];
+
+  buildInputs = [
+    fftwSinglePrec
+    ocl-icd
+    opencl-headers
+  ] ++ lib.optional enableCuda cudatoolkit
+  ;
+
+  propagatedBuildInputs = [
+    python
+    numpy
+  ];
+
+  dontUseSetuptoolsBuild = true;
+  dontUsePipInstall = true;
+  dontUseSetuptoolsCheck = true;
+
+  cmakeFlags = [
+    "-DCMAKE_C_COMPILER=${clang}/bin/clang"
+    "-DCMAKE_CXX_COMPILER=${clang}/bin/clang++"
+    "-DBUILD_TESTING=ON"
+    "-DOPENMM_BUILD_AMOEBA_PLUGIN=ON"
+    "-DOPENMM_BUILD_CPU_LIB=ON"
+    "-DOPENMM_BUILD_C_AND_FORTRAN_WRAPPERS=ON"
+    "-DOPENMM_BUILD_DRUDE_PLUGIN=ON"
+    "-DOPENMM_BUILD_PME_PLUGIN=ON"
+    "-DOPENMM_BUILD_PYTHON_WRAPPERS=ON"
+    "-DOPENMM_BUILD_RPMD_PLUGIN=ON"
+    "-DOPENMM_BUILD_SHARED_LIB=ON"
+    "-DOPENMM_BUILD_AMOEBA_OPENCL_LIB=ON"
+    "-DOPENMM_BUILD_OPENCL_LIB=ON"
+    "-DOPENMM_BUILD_DRUDE_OPENCL_LIB=ON"
+    "-DOPENMM_BUILD_RPMD_OPENCL_LIB=ON"
+  ] ++ lib.optionals enableCuda [
+    "-DCUDA_SDK_ROOT_DIR=${cudatoolkit}"
+    "-DOPENMM_BUILD_AMOEBA_CUDA_LIB=ON"
+    "-DOPENMM_BUILD_CUDA_LIB=ON"
+    "-DOPENMM_BUILD_DRUDE_CUDA_LIB=ON"
+    "-DOPENMM_BUILD_RPMD_CUDA_LIB=ON"
+    "-DCMAKE_LIBRARY_PATH=${cudatoolkit}/lib64/stubs"
+  ];
+
+  enableParallelBuilding = true;
+
+  # "make PythonInstall" wants to install into the store-path of the python executable.
+  # Therefore, execute setup.py manually and choose the appropriate prefix.
+  postInstall = ''
+    cd python
+    export OPENMM_INCLUDE_PATH=$out/include
+    export OPENMM_LIB_PATH=$out/lib
+    python setup.py build && python setup.py install --prefix=$out
+  '';
+
+  # Needs to explicitly set PYTHONPATH when using.
+  #pythonImportsCheck = [ "openmm.app" ];
```

Comment:
What's this for?

Reply:
See my initial message in the PR.

>The package works when explicitly setting PYTHONPATH but cannot be imported otherwise. This is also the reason why the pythonImportsCheck is currently commented out. I couldn't figure out what the problem is and appreciate any idea :)

Reply:
Hmm, so basically we break either binary usage or importing?

Reply:
Could you elaborate how these two relate? Why does binary usage break importing or vice versa? I feel I am missing some knowledge here.

Reply:
I'm as confused here as you are. I guess we need some more eyes to look at this.

Reply:
Is there any confusion on the module name due to the presence of a folder in the working directory of something?

Reply:
> Is there any confusion on the module name due to the presence of a folder in the working directory of something?

No I don't think so. Happens even from an empty directory and the directory structure of the result looks perfectly normal:

```
result/lib/
├── libOpenMMAmoeba.so
├── libOpenMMDrude.so
├── libOpenMMRPMD.so
├── libOpenMM.so -> libOpenMM.so.7.7
├── libOpenMM.so.7.7
├── plugins
│   ├── libOpenMMAmoebaOpenCL.so
│   ├── libOpenMMAmoebaReference.so
│   ├── libOpenMMCPU.so
│   ├── libOpenMMDrudeOpenCL.so
│   ├── libOpenMMDrudeReference.so
│   ├── libOpenMMOpenCL.so
│   ├── libOpenMMPME.so
│   ├── libOpenMMRPMDOpenCL.so
│   └── libOpenMMRPMDReference.so
└── python3.10
    └── site-packages
        ├── openmm
        ├── OpenMM-7.7.0-py3.10.egg-info
        └── simtk
``` 



> If the python package can't be imported it is likely broken and does not work correct.

Works perfectly as expected when one sets the PYTHONPATH. That is not the problem.

Reply:
Alright, this requires investigation then.

Reply:
OK, I am building now with stdenv.mkDerivation and convert with `toPythonModule` afterwards. Looks more clean. I can also import openmm from a nix-shell. A test from within the derivation somehow seems to suffer from an import of a swig-generated interface. 
`result/lib/python3.10/site-packages/openmm/_openmm*.so` is not found before the derivation is finalised. However, the finished derivation can be imported from Python without problems.